File: 10 平常练习/Xocde 9/Xocde 9/ViewController.m
//
//  ViewController.m
//  Xocde 9
//
//  Created by JQ on 2017/9/21.
//  Copyright © 2017年 Majq. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    int a ;
    
    int *p = &a;
    
    *p = 3;
   
    
    NSLog(@"%d",a);
    
    int b1 = 2;
    int b2 = 3;
    
    swapT(&b1, &b2);
    
    NSLog(@"%d %d",b1,b2);
    
}

void swapT(int *a, int *b) {
    int t = *a;
    
    *a = *b;
    
   * b = t;
    
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 10 平常练习/Swift 4.0/Swift 4.0/ViewController.swift
//
//  ViewController.swift
//  Swift 4.0
//
//  Created by JQ on 2017/9/20.
//  Copyright © 2017年 Majq. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
      
       
        let p = Person(fullyName: "jack")
        print(p.fullyName)
        
        
       
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

protocol FullyName {
    var fullyName: String {get}
}

protocol Run {
    var run: String {set get}
    
}

struct Person: FullyName {
    var fullyName: String
    
    
    
}


